tests/src/test/java/org/eclipse/hono/tests/coap/CoapTestBase.java

```diff
@@ -408,6 +410,56 @@ public void testUploadMessagesViaGateway(final VertxTestContext ctx) throws Inte
                 });
     }
 
+    /**
+     * Verifies that an edge device is auto-provisioned if it connects via a gateway equipped with the corresponding
+     * authority.
+     *
+     * @param ctx The test context.
+     * @throws InterruptedException if the test fails.
+     */
+    @Test
+    public void testAutoProvisioningViaGateway(final VertxTestContext ctx) throws InterruptedException {
+
+        // GIVEN a device that is connected via two gateways
+        final Tenant tenant = new Tenant();
+        final String gatewayId = helper.getRandomDeviceId(tenantId);
+        final Device gateway = new Device()
+                .setAuthorities(Collections.singleton(RegistryManagementConstants.AUTHORITY_AUTO_PROVISIONING_ENABLED));
+
+        final VertxTestContext setup = new VertxTestContext();
+        helper.registry.addPskDeviceForTenant(tenantId, tenant, gatewayId, gateway, SECRET)
+                .onComplete(setup.completing());
+        ctx.verify(() -> assertThat(setup.awaitCompletion(5, TimeUnit.SECONDS)).isTrue());
+
+        final CoapClient gatewayClient = getCoapsClient(gatewayId, tenantId, SECRET);
+
+        final VertxTestContext emptyEventReceived = new VertxTestContext();
+        helper.applicationClientFactory.createEventConsumer(tenantId,
+                msg -> emptyEventReceived.verify( () -> {
+                    // ignore potential events of warmup
+                    if (MessageHelper.getDeviceId(msg).equals(deviceId)) {
+                        assertThat(msg.getContentType()).isEqualTo(EventConstants.CONTENT_TYPE_EMPTY_NOTIFICATION);
+                        assertThat(MessageHelper.getRegistrationStatus(msg)).isEqualTo(EventConstants.RegistrationStatus.NEW.name());
+                        emptyEventReceived.completeNow();
+                    }
+                }),
+                close -> {});
+
+        testUploadMessages(ctx, tenantId,
```

Comment:
IMHO sending the message could/should be done in a much simpler way, e.g. as in the *testUploadMessageFailsForDisabledDevice* test method ...